docs/fml-spec.md

```diff
@@ -0,0 +1,576 @@
+  ThumbnailType:
+    description: An enum of types of icon that can be presented alongside titles of pages
+    variants:
+      letter: A rendering of the first letter of the domain name
+      screenshot: A screenshot thumbnail of the webppage
+      favicon: The favicon derived from the webpage
+      none: No icon is displayed
+```
+
+The value `itemThumbnail` is of type `ThumbnailType`, which is generated as an enum. This can be used exhaustively matched in a `switch` statement.
+
+```swift
+switch spotlightConfig.itemThumbnail {
+    case .favicon:
+        item.thumbnailData = FaviconFetcher.getFaviconFromDiskCache(imageKey: baseDomain)?.pngData()
+    case .letter:
+        item.thumbnailData = FaviconFetcher.letter(forUrl: url).pngData()
+    case .screenshot:
+        item.thumbnailData = tab.screenshot?.pngData()
+    case .none:
+        item.thumbnailData = nil
+}
+```
+
+Note that enumeration variants in Swift are in `mixedCamelCase`; in Kotlin they are in `SCREAMING_SNAKE_CASE`.
+
+## Feature defaults
+
+Already, in this simple example we have three variables. We may want to vary the configuration of the feature when the user is not involved in an experiment.
+
+This might be because we have run an experiment and learned that a change of configuration should be made more permanent.
+
+The `defaults` list is a list of zero or more "default blocks", used to patch the default values of the variables in a feature.
+
+The `value` property defines the patch which is overlaid on top of the existing variable values.
+
+```yaml
+features:
+  spotlight-search:
+    variables:
+      enabled:
+        description: If `false`, the app will not record anything with Spotlight.
+        type: Bool
+        default: false
+      max-age-in-days:
+        description: The number of days a single piece of content is indexed for.
+        type: Double
+        default: 28.0
+      item-thumbnail:
+        description: "The icon that appears in the Spotlight search results.
+          Note that changing this does not change already indexed content."
+        type: ThumbnailType
+        default: letter
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+```
+
+The `defaults` list of `value`s set the default value of `item-thumbnail` to `screenshot`, and `max-age-in-days` to `56.0`.
+
+Adding new feature blocks to the `defaults` list patches the existing defaults further:
+
+```yaml
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+      - value: { max-age-in-days: 64.0 }
+```
+
+In this case, the default value for `item-thumbnail` is now `screenshot`, and `max-age-in-days` is `64.0`.
+
+### Feature defaults and channels
+
+For release engineering reasons, or marketing reasons, we might want to ensure that the feature isn't enabled by default for the release population of users.
+
+However, for quality reasons, we might want a different configuration for our nightly audience.
```

Comment:
This explanation around "engineering/marking/quality" is a little weird – I would frame this more as something like 

"Sometimes it is useful to have different default configuration for different *channels* of an application. For example, you may want to have a feature available for testing on Nightly or Beta before turning it on in Release".

Reply:
Nice!